arbitrator/jit/src/wavmio.rs

```diff
@@ -101,6 +109,38 @@ pub fn read_delayed_inbox_message(mut env: WasmEnvMut, sp: u32) -> MaybeEscape {
     inbox_message_impl(&sp, inbox, "wavmio.readDelayedInboxMessage")
 }
 
+// Reads a hotshot commitment
+fn read_hotshot_commitment_impl(
+    sp: &GoStack,
+    comm_map: &HotShotCommitmentMap,
+    name: &str,
+) -> MaybeEscape {
+    let msg_num = sp.read_u64(0);
+    let out_ptr = sp.read_u64(1);
+    let out_len = sp.read_u64(2);
+    if out_len != 32 {
+        eprintln!("Go trying to read header bytes with out len {out_len} in {name}");
+        sp.write_u64(5, 0);
+        return Ok(());
+    }
+
+    macro_rules! error {
+        ($text:expr $(,$args:expr)*) => {{
+            let text = format!($text $(,$args)*);
+            return Escape::hostio(&text)
+        }};
+    }
+
+    let message = comm_map.get(&msg_num).unwrap_or(&[0; 32]);
+
+    if out_ptr + 32 > sp.memory_size() {
+        error!("memory bounds exceeded in {name}");
```

Comment:
macro `error!` is used only for once, I think it is better to expand it here and remove the macro.